accepted/2021/system-drawing-win-only/system-drawing-win-only.md

```diff
@@ -0,0 +1,193 @@
+effectively unmaintained.
+
+While `libgdiplus` has helped a lot of customers expand to cross-platform it has
+also caused some headaches. `System.Drawing.Common` is very Windows driven as it
+is mostly a wrapper around GDI+ and is heavily coupled to Windows Forms, which
+keeps evolving and diverging from cross-platform graphics. This has caused some
+confusion over why some things don't work well on Unix or don't work at all.
+
+Since the inclusion of the Mono cross-platform implementation we spent lot of
+time redirecting issues to `libgdiplus` that never got fixed, or helping people
+install `libgdiplus` correctly (because it's not part of .NET Core proper). We
+have thought about our options, and to support `System.Drawing.Common` properly
+on Unix we'd need to make `libgdiplus` a first-class citizen and ship it as part
+of the product:
+
+* `libgdiplus` doesn't ship with all operating systems by default, just like
+  other native libraries we depend on (icu, openssl).
+
+* `libgdiplus` is community driven, so a lot of issues/PRs just go stale and are
+  never looked at.
+
+* There is no servicing story for bugs and security issues.
+
+However, making `libgdiplus` a first-class citizen is costly. It's around 30k
+lines of pretty old C code, virtually untested, and lacks a lot of
+functionality. `libgdiplus` also has a lot of external dependencies for all the
+image processing and text rendering, such as `cairo`, `pango`, and other native
+libraries, which makes shipping even more challenging.
+
+We've also hit some memory leaks or double free'd memory that weren't fixed in
+`libgdiplus` and that we had to work around because they were potential security
+issues. This makes it harder as we are basically on our “own” for those security
+issues. `libgdiplus` is a library we don't own, but that we must spend time
+investigating issues and proposing fixes.
+
+We have noticed that `System.Drawing.Common` is used on cross-platform mostly for
+image manipulation like QR code generators and text rendering. We haven't
+noticed heavy graphics usage as our cross-platform graphics support is very
+incomplete. For example, right now some components crash on macOS because the
+native APIs we use have been deprecated and are no longer included with recent
+macOS releases. We have never gotten a report about that crash; we discovered it
+ourselves while making `System.Drawing.Common` trim compatible.
+
+The usages we see of `System.Drawing.Common` in non-Windows environments are
+typically well supported with SkiaSharp and ImageSharp.
+
+Thus, we believe it is not worth investing in making `System.Drawing.Common` work
+on non-Windows platforms and instead propose we only keep evolving it in the
+context of Windows Forms and GDI+.
+
+## Requirements
+
+### Goals
+
+* Consumers of `System.Drawing.Common` will be informed that starting with .NET
+  6, it's only supported on Windows, via the platform compatibility analyzer
```

Comment:
```suggestion
  6, it's only supported on Windows, via the [platform compatibility analyzer](https://docs.microsoft.com/en-us/dotnet/standard/analyzers/platform-compat-analyzer)
```

BTW, are we supposed to use hard breaks in markdown? I haven't been..

Reply:
Yes, because Markdown is supposed to be readable in clear text and unwrapped lines aren't friendly in this regard.

Reply:
Ugh -- which text editor doesn't have a wrapping toggle. But, OK.